scripts/scripts.js

```diff
@@ -396,6 +396,101 @@ async function loadEager(doc) {
   }
 }
 
+function videoEventTracking(myPlayer, videoTagId) {
+  let eventName = 'video_started';
+  let progress = [];
+
+  myPlayer.on('timeupdate', (event) => {
+    const videoPercentage = (myPlayer.currentTime() * 100) / myPlayer.duration();
+
+    switch (true) {
+      case videoPercentage === 100:
+        eventName = 'video_completed';
+        break;
+      case videoPercentage > 90:
+        eventName = 'video_reached_90_pct';
+        break;
+      case videoPercentage > 75:
+        eventName = 'video_reached_75_pct';
+        break;
+      case videoPercentage > 50:
+        eventName = 'video_reached_50_pct';
+        break;
+      case videoPercentage > 25:
+        eventName = 'video_reached_25_pct';
+        break;
+      default:
+        eventName = 'video_started';
+    }
+
+    if (!progress.includes(eventName)) {
+      progress.push(eventName);
+
+      const videoProgress = {
+        name: eventName,
+        digitalData: {
+          videoTracking: {
+            id: videoTagId,
+            name: myPlayer.mediainfo.name,
+          },
+        },
+        event,
+      };
+
+      if (eventName === 'video_completed') {
+        progress = [];
+      }
+
+      window.appEventData.push(videoProgress);
```

Comment:
Maybe here same as https://github.com/hlxsites/servicenow/pull/109/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR458 before pushing, do `window.appEventData = window.appEventData  || []`, in case no one else initialised it.

Reply:
good catch.